app/src/main/java/com/asfoundation/wallet/home/usecases/GetImpressionUseCase.kt

```diff
@@ -0,0 +1,17 @@
+package com.asfoundation.wallet.home.usecases
+
+import com.appcoins.wallet.core.utils.android_common.RxSchedulers
+import com.asfoundation.wallet.repository.ImpressionRepository
+import io.reactivex.Completable
+import javax.inject.Inject
+
+class GetImpressionUseCase
+@Inject
+constructor(
+  private val impressionRepository: ImpressionRepository,
+  private val schedulers: RxSchedulers
+) {
+  operator fun invoke(): Completable {
+    return impressionRepository.getImpression().observeOn(schedulers.io)
```

Comment:
this should be "subscribeOn". the observeOn should be on main thread.
Anyway, the best option should be to leave it only on the repository

Reply:
I made the change to subscribeOn, but I don't think the part about using only the repository should be changed, because we're trying to keep the viewModel standard of importing the Usecase and not the repository.